### PR TITLE
fix found issues of comm replay

### PR DESCRIPTION
**Pull request by shengfukevin**
## Summary
fix replay to send/recv
fix blocking option
fix process group update of barrier
Comment out unused function incast and multicast
remove function extractCommsInfo() for deprecatd basic trace
fix wait comm op for both collective and p2p

## Test Plan
comm_replay --trace-type et --trace-path /home/sanshang/lustre_storage/000_code/chakra/comm_replay_eval_06/training/et_iter_50_51 --num-replays 10

## additional notes
This is copied from https://github.com/facebookresearch/param/pull/180.

### CODE REVIEW

File: et_replay/comm/backend/base_backend.py [hunk 1]
```diff
@@ -102,7 +100,9 @@ def __init__(self) -> None:
         self.dataSize = 0
         self.numElements = 0
         self.waitObj = []
-        self.waitObjIds = {}  # mapping of reqID to future of async collectives
+        self.waitObjIds = (
+            {}
+        )  # mapping of (pg_id, req_id, is_p2p) to future of async collectives
```
Comment (GSSBMW) [inline et_replay/comm/backend/base_backend.py:105]: No parenthesis here.
image

File: et_replay/tools/comm_replay.py [hunk 2]
```diff
@@ -802,9 +802,15 @@ def runComms(
...
-                # record collectiveID for wait ops
-                if curComm.req is not None:
-                    self.collectiveArgs.collectiveId = curComm.req
+                # record wait_obj_key for wait ops
+                if curComm.req is not None and curComm.pgId is not None:
+                    self.collectiveArgs.wait_obj_key = (
+                        curComm.pgId,
+                        curComm.req[0],
+                        curComm.req[1],
+                    )
```
Comment (GSSBMW) [inline et_replay/tools/comm_replay.py:811]: changed by lint rules?
image

File: et_replay/tools/comm_replay.py [hunk 3]
```diff
@@ -832,10 +838,16 @@ def runComms(
...
-            if curComm.req is not None and not self.is_blocking and collName != "wait":
-                self.collectiveArgs.waitObjIds[curComm.req] = retObj
+            if (
+                not self.is_blocking
+                and collName != "wait"
+                and self.collectiveArgs.wait_obj_key is not None
+            ):
+                self.collectiveArgs.waitObjIds[self.collectiveArgs.wait_obj_key] = (
+                    retObj
+                )
```
Comment (GSSBMW) [inline et_replay/tools/comm_replay.py:850]: Is parenthesis at 843 and 848 necessary?
image

**Comment by shengfukevin** [comment]
@GSSBMW, the parenthesis is added by lint, basically to split one line into multiple lines. It does not change the syntax.

**Comment by GSSBMW** [comment]
@shengfukevin Ok. Then please check several ones I comment, whether it's necessary. Especially, only one element in the added parentheses. Thanks!

**Comment by shengfukevin** [comment]
The code change you commented on are all done by lint. So it should be ok.

**Comment by GSSBMW** [comment]
Then please merge. Thanks!

**Pull request closed without merging**